File: src/main/java/com/company/erp/features/users/dto/CreateUserRequest.java
package com.company.erp.features.users.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import java.util.Set;

public record CreateUserRequest(
        @NotBlank @Email String email,
        @NotBlank @Size(min = 8, max = 100) String password,
        @NotBlank @Size(max = 255) String fullName,
        @Size(max = 50) String phone,
        /** Role codes to assign (e.g. ["STAFF"]). Optional. */
        Set<String> roles
) {
    public CreateUserRequest {
        if (roles == null) roles = Set.of();
    }
}

File: app/Features/Users/Requests/CreateUserRequest.php
<?php

namespace App\Features\Users\Requests;

use App\Support\Http\ApiFormRequest;

class CreateUserRequest extends ApiFormRequest
{
    public function rules(): array
    {
        return [
            'email' => ['required', 'email', 'max:255'],
            'password' => ['required', 'string', 'min:8', 'max:100'],
            'fullName' => ['required', 'string', 'max:255'],
            'phone' => ['nullable', 'string', 'max:50'],
            'roles' => ['nullable', 'array'],
            'roles.*' => ['string'],
        ];
    }

    /**
     * @return array<int, string>
     */
    public function roleCodes(): array
    {
        return array_values(array_unique($this->input('roles', []) ?: []));
    }
}
